lib/tasks/eu_exit_business_finder.rake

```diff
@@ -0,0 +1,78 @@
+namespace :eu_exit_business_finder do
+  task update_subscriber_list_titles_and_facet_values: [:environment] do
+    subscriber_lists = SubscriberList
+      .where("links::text != '{}'")
+      .where("ARRAY(SELECT json_object_keys(links)) <@ Array[:keys]", keys: [:facet_values])
+    update_facet_values(subscriber_lists)
+  end
+
+  # Iterate through the affected subscriber lists and if we encounter any
+  # facet values that need replacing with a newer one, we insert the new one
+  # to the array (inserted 'in place' to retain original order). Then remove
+  # the old facet values in bulk and write only the unique values that remain.
+  def update_facet_values(subscriber_lists)
+    subscriber_lists.each do |list|
+      links = list.links[:facet_values][:any]
+      updated_links = links.dup # copy the links to avoid corrupting the loop
+      links.each_with_index do |value, index|
```

Comment:
I wonder if `map` could be utilised to remove the need to create `updated_links`, remove the `each` loop and remove the `insert`?  For example (I haven't tested it, but something like this maybe):
```
links.map { |link|
  EuExitFacetMigrationConfig::facet_values_to_replace.find { |replace_val| replace_val[:old_facet_value] == value }.fetch(:new_facet_value, nil) || link
}
```